### ISSUE TITLE
Automapper 5.0 - AutoMapperMappingException when mapping arrays to lists

### ISSUE THREAD

**Issue by tpodolak** [AutoMapper/AutoMapper#1424]
I have following classes
```
public class FlightViewModel
		{
			public List<JourneyViewModel> Journeys { get; set; }
		}
		public class FlightModel
		{
			public Journey[] Journeys { get; set; }
		}
		public class Journey
		{
		}
		public class JourneyViewModel
		{
		}
```
In AutoMapper 4.2.1.0 I was able to map model with null array of ``Journeys``
```
var model = new FlightModel();
var result = mapper.Map<FlightModel, FlightViewModel>(model);
```
by using default configuration
```
configuration.CreateMissingTypeMaps = true;
configuration.AllowNullCollections = true;
configuration.AllowNullDestinationValues = true;
configuration.CreateMap<FlightModel, FlightViewModel>();
configuration.CreateMap<Journey, JourneyViewModel>();
```
after switching to Automapper 5.0 the same code throws 
```
AutoMapper.AutoMapperMappingException: Error mapping types.

Mapping types:
FlightModel -> FlightViewModel
ConsoleApplication1.Program+FlightModel -> ConsoleApplication1.Program+FlightViewModel

Type Map configuration:
FlightModel -> FlightViewModel
ConsoleApplication1.Program+FlightModel -> ConsoleApplication1.Program+FlightViewModel

Property:
Journeys ---> System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Cast[TResult](IEnumerable source)
   at lambda_method(Closure , FlightModel , FlightViewModel , ResolutionContext )
   --- End of inner exception stack trace ---
   at lambda_method(Closure , FlightModel , FlightViewModel , ResolutionContext )
   at AutoMapper.Mapper.AutoMapper.IMapper.Map[TSource,TDestination](TSource source)
   at ConsoleApplication1.Program.Main(String[] args) in C:\Users\podolakt\Documents\Visual Studio 2015\Projects\ConsoleApplication1\ConsoleApplication1\Program.cs:line 23
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
   at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```
Is there any global switch to restore old behaviour? I have quite significant amount of these kind of situations in my project so manually settings maps for every scenario is not an option

**Comment by lbargaoanu** [comment]
This is a bug. A quick workaround is to make both properties arrays or lists (or smth else).

**Comment by tpodolak** [comment]
@lbargaoanu unfortunately I can't do that as it would affect too many projects(changes in shared library would be required) so I would have to wait for bugfix

**Comment by jbogard** [comment]
"CreateMissingTypeMaps" noooooooo

**Comment by lbargaoanu** [comment]
@jbogard Yes, but it doesn't seem relevant. The bug happens anyway.

**Comment by lbargaoanu** [comment]
@tpodolak You can try the MyGet build and see if it solves your problem.

**Comment by jigneshkvp** [comment]
Hi,

Is this fix included in the release 5.0.2?

**Comment by tpodolak** [comment]
@jigneshkvp yes, works for me.
